src/plugins/ec.ts

```diff
@@ -172,6 +173,30 @@ export class ECPlugin extends BasePlugin {
             }, {});
     }
 
+    private convertProductPriceToNumber(product: Product) {
+        if (product.price) {
+            product.price = this.tryConvertStringPriceToNumber(product.price);
+        }
+
+        return product;
+    }
+
+    private convertImpressionPriceToNumber(impression: BaseImpression) {
+        if (impression.price) {
+            impression.price = this.tryConvertStringPriceToNumber(impression.price);
+        }
+
+        return impression;
+    }
+
+    private tryConvertStringPriceToNumber(price: number | string): number | string {
+        if (typeof price === 'number') {
+            return price;
+        }
+
+        return Number(price.replace(/[^0-9\.-]/g, '')) || price;
```

Comment:
I think we unfortunately have to handle a number being zero. Otherwise, if the number is zero, it'll fallback to `price` 🤔. Sorry, should've caught that sooner.
```suggestion
        const parsedNumber = Number(price.replace(/[^0-9\.-]/g, ''));
        return parsedNumber != 0 ? parsedNumber : price;
```

Reply:
I'll add a test case